File: F/spiral.f
C***********************************************************----------!!
	SUBROUTINE spiral(x, y, npts, xnot, ynot,turns, alpha)
	INTEGER npts, xnot, ynot, x(npts), y(npts)
	REAL turns,alpha
	
	x(1) = xnot
	y(1) = ynot
	DO 1000 i = 2, npts
	  x(i) = INT( xnot + FLOAT(i)*alpha*
     1         cos(2.*3.14159*turns*FLOAT(i)/FLOAT(npts-1)) )
	  y(i) = INT( ynot + FLOAT(i)*alpha*
     1         sin(2.*3.14159*turns*FLOAT(i)/FLOAT(npts-1)) )
CX	  PRINT *,x(i), y(i)
 1000 CONTINUE
      RETURN
      END

File: C/spiral.c
#include <math.h>

void spiral(px, py, npts, xnot, ynot,turns, alpha)
int npts, xnot, ynot, *px, *py;
float turns, alpha;
{ 
  int i, *pxt, *pyt;
  float pi;
  
  pxt = px;
  pyt = py;
  
  *pxt = xnot;
  *pyt = ynot;
  pxt++;
  pyt++;
	pi = acos(-1.0);
	for (i = 1; i< npts; i++)
	 {
	  *pxt = xnot + i*alpha*cos(2.*pi*turns*i/(npts-1));
	  *pyt = ynot + i*alpha*sin(2.*pi*turns*i/(npts-1));
	  pxt++;
	  pyt++;
       }

}

